### PR TITLE
Refactor for multiple dbs, firewall, vnet rules

### PR DESCRIPTION
**Pull request by sdickenson**
Make resource group creation optional

This breaks heavily anyone using v2. Didn't update tests or documentation; will do that in a later revision.

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -8,11 +8,6 @@ output "server_name" {
   value       = azurerm_mysql_server.mysql.name
 }
 
-output "database_name" {
-  description = "The database name of MySQL Server."
-  value       = azurerm_mysql_database.mysql.name
-}
```
Comment (sblack4) [inline outputs.tf:14]: why take this out?

**Review by sblack4 — APPROVED** [review]
:shipit:

File: outputs.tf (hunk 1 above)
Comment (sdickenson) [reply outputs.tf:14]: Because I'm using a map now to create multiple databases, thus the dbs var.

Comment (sblack4) [reply outputs.tf:14]: ohh yeah that'll mess up the output that I don't know if we need. I think that makes this a major release?

File: main.tf [hunk 2]
```diff
@@ -1,13 +1,18 @@
+provider "azurerm" {
+  version = ">=1.40.0"
```
Comment (sblack4) [inline main.tf:2]: This will cause problems if we're using a different provider version

Comment (sdickenson) [reply main.tf:2]: I can knock it down to a lesser version, but I haven't tested it on those.

File: outputs.tf (hunk 1 above)
Comment (sdickenson) [reply outputs.tf:14]: Yeah, I was tagging this 3.0.0

**Pull request merged**